site/pages/create-consequence-services/[disruptionId]/[consequenceIndex].page.tsx

```diff
@@ -54,9 +54,25 @@ const filterConfig = {
     matchFrom: "any" as const,
 };
 
-export const fetchStops = async (serviceId: number): Promise<Stop[]> => {
+export const fetchStops = async (
+    serviceId: number,
+    vehicleMode?: VehicleMode | Modes,
+    dataSource?: Datasource,
+): Promise<Stop[]> => {
     if (serviceId) {
-        const stopsData = await fetchServiceStops({ serviceId });
+        const stopsData = await fetchServiceStops({
+            serviceId,
+            modes: vehicleMode === VehicleMode.tram ? "tram, metro" : vehicleMode,
+            ...(vehicleMode === VehicleMode.bus ? { busStopType: "MKD" } : {}),
+            ...(vehicleMode === VehicleMode.bus
+                ? { stopTypes: "BCT" }
+                : vehicleMode === VehicleMode.tram || vehicleMode === Modes.metro
+                ? { stopTypes: "MET, PLT" }
+                : vehicleMode === Modes.ferry || vehicleMode === VehicleMode.ferryService
+                ? { stopTypes: "FER, FBT" }
+                : { stopTypes: "undefined" }),
```

Comment:
No data is returned for trains because stopTypes is set to undefined string

Reply:
as mentioned above